packages/graphql-yoga/__tests__/recipes.spec.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  createSchema,
+  createYoga,
+  GraphQLParams,
+  YogaInitialContext,
+} from 'graphql-yoga'
+
+describe('recipe', () => {
+  it('id as custom top level POST body query parameter', async () => {
+    const yoga = createYoga({
+      schema: createSchema({
+        typeDefs: /* GraphQL */ `
+          type Query {
+            id: String
+          }
+        `,
+        resolvers: {
+          Query: {
+            id: (_, __, context: YogaInitialContext) => {
+              return context.params.extensions?.id
+            },
+          },
+        },
+      }),
+      plugins: [
+        {
+          /**
+           * Plugin for allowing the client to send the query ID as a custom POST body query parameter.
+           * Before the query parameter validation is happening it is moved to the extensions object.
+           */
+          onRequestParse() {
+            return {
+              onRequestParseDone(ctx) {
+                function process(requestParserResult: GraphQLParams) {
+                  if ('id' in requestParserResult) {
+                    return {
+                      ...Object.fromEntries(
+                        Object.entries(requestParserResult).filter(
+                          ([key]) => key !== 'id',
+                        ),
+                      ),
```

Comment:
@ardatan Maybe we should make the parameter check more forgiving by allowing `undefined` as a value, and thus avoid this mutation?

Reply:
I'm fine with that. `!= null` should do the trick.

Reply:
https://github.com/dotansimha/graphql-yoga/blob/main/packages/plugins/persisted-operations/src/index.ts#L78
Maybe you can use `onParams` like here because it seems to be working fine with non-standard `doc_id` in tests;
https://github.com/dotansimha/graphql-yoga/blob/main/packages/plugins/persisted-operations/__tests__/persisted-operations.spec.ts#L197

In that case, you don't need to manipulate different request parser results.